packages/webapp/src/elections/api/eden-contract.ts

```diff
@@ -1,33 +1,163 @@
+    }
 
-    if (!rows.length) {
+    // TODO: see what the actual data looks like and see if we need to do this +1 to match up 0-based and 1-based data
+    return {
+        groupNumber: groupNumber + 1,
+        lowerBound: lowerBound + 1,
+        upperBound: upperBound + 1,
+    };
+};
+
+export const getMemberGroupParticipants = async (
+    memberAccount: string | undefined,
+    config: ActiveStateConfigType
+) => {
+    if (!memberAccount || !config) return undefined;
+
+    const totalParticipants = config.num_participants;
+    const numGroups = config.num_groups;
+
+    // get member index
+    const memberVoteData = await getVoteDataRow({
+        fieldValue: memberAccount,
+    });
+    if (!memberVoteData) return [];
+
+    // return all indexes that represent members in this member's group
+    const { groupNumber, lowerBound, upperBound } = getMemberGroupFromIndex(
+        // TODO: remove this -1 if no conversion is needed between 0=based and 1=based arrays
+        memberVoteData.index - 1,
+        totalParticipants,
+        numGroups
+    );
+
+    // get all members in this member's group
+    const rows = await getVoteDataRows({
+        lowerBound,
+        upperBound,
+    });
+
+    if (!rows || !rows.length) {
         return undefined;
     }
 
-    return rows[0];
+    return rows;
 };
 
-export const getElectionState = async () => {
-    if (devUseFixtureData) return fixtureElectionState;
+const getVoteDataRow = async (
+    opts: VoteDataQueryOptionsByField
+): Promise<VoteData | undefined> => {
+    if (devUseFixtureData)
+        return Promise.resolve(fixtureVoteDataRow(opts.fieldValue));
 
-    const rows = await getTableRows<ElectionState>(
-        CONTRACT_ELECTION_STATE_TABLE
+    const memberVoteData = await getRow<VoteData>(
+        CONTRACT_VOTE_TABLE,
+        opts.fieldName,
+        opts.fieldValue
     );
+    return memberVoteData;
+};
+
+const getVoteDataRows = async (
+    opts: VoteDataQueryOptionsByGroup
+): Promise<VoteData[] | undefined> => {
+    if (devUseFixtureData)
+        return Promise.resolve(
+            fixtureVoteDataRows.filter(
+                (vote) =>
+                    vote.index >= opts.lowerBound &&
+                    vote.index <= opts.upperBound
+            )
+        );
+
+    // TODO: see what real data looks like and real use-cases and see if we need the electionState flag;
+    // If not, switch this back to getTableRows()
+    const rawRows = await getTableRawRows<VoteData>(CONTRACT_VOTE_TABLE, opts);
+    // const electionState = rawRows[0][0];
+
+    const rows = rawRows.map((row) => row[1]);
 
     if (!rows.length) {
         return undefined;
     }
 
-    return rows[0];
+    return rows;
+};
+
+export const getCurrentElection = async () => {
+    if (devUseFixtureData) return fixtureCurrentElection;
+
+    const rawRows = await getTableRawRows<any>(CONTRACT_CURRENT_ELECTION_TABLE);
+    const electionState = rawRows[0][0];
```

Comment:
What is the data inside the electionState? I suspect its the variant struct name, like `election_st_v0`, and if so why do we need it? Maybe I missed the usage?

Reply:
Not sure I understand the question @sparkplug0025 . That *is* the variant struct name. We need this data for election meta info like round start/end time and current election status, eg. seeding, active, final, etc.

Reply:
Yeah, I'm curious to know what you will do with the structure name in the `electionState` variable, I didn't see any usage for it. I'm asking that from the view that all of our other contract table selections we don't do anything with the variant struct name, and then I'm curious to know what you will do with it.

Reply:
Yeah, the shape of the value that the variant holds is *dependent on* the variant type, and this type (which indicates election status) isn't represented elsewhere, so I've gotta pull it from here so we know election state